src/measure_theory/l1_space.lean

```diff
@@ -157,15 +176,139 @@ lemma integrable_sub {f g : α → β} (hf : measurable f) (hg : measurable g) :
   by { rw sub_eq_add_neg, refine integrable_add hf (measurable_neg hg) hfi (integrable_neg hgi) }
 
 lemma integrable_norm {f : α → β} (hfi : integrable f) : integrable (λa, ∥f a∥) :=
-calc (∫⁻ (a : α), (nnnorm ∥f a∥)) = (∫⁻ (a : α), (nnnorm (f a))) :
-    begin
-      apply lintegral_congr_ae,
-      filter_upwards [],
-      assume a,
-      simp only [mem_set_of_eq],
-      rw [nnnorm_norm]
-    end
-  ... < ⊤ : hfi
+have eq : (λa, (nnnorm ∥f a∥ : ennreal)) = λa, (nnnorm (f a) : ennreal),
+   by { funext, rw nnnorm_norm },
+ by { rwa [integrable, eq] }
+
+ lemma integrable_norm_iff (f : α → β) : integrable (λa, ∥f a∥) ↔ integrable f :=
+ have eq : (λa, (nnnorm ∥f a∥ : ennreal)) = λa, (nnnorm (f a) : ennreal),
+   by { funext, rw nnnorm_norm },
+ by { rw [integrable, integrable, eq] }
+
+ lemma integrable_of_integrable_bound {f : α → β} {bound : α → ℝ} (h : integrable bound)
+   (h_bound : ∀ₘ a, ∥f a∥ ≤ bound a) : integrable f :=
+ have h₁ : ∀ₘ a, (nnnorm (f a) : ennreal) ≤ ennreal.of_real (bound a),
+ begin
+   filter_upwards [h_bound],
+   simp only [mem_set_of_eq],
+   assume a h,
+   calc (nnnorm (f a) : ennreal) = ennreal.of_real (∥f a∥) : by rw of_real_norm_eq_coe_nnnorm
+     ... ≤ ennreal.of_real (bound a) : ennreal.of_real_le_of_real h
+ end,
+ calc (∫⁻ a, nnnorm (f a)) ≤ (∫⁻ a, ennreal.of_real (bound a)) :
+     by { apply lintegral_le_lintegral_ae, exact h₁ }
+   ... ≤ (∫⁻ a, ennreal.of_real ∥bound a∥) : lintegral_le_lintegral _ _ $
+     by { assume a, apply ennreal.of_real_le_of_real, exact le_max_left (bound a) (-bound a) }
+   ... < ⊤ : by { rwa [integrable_def_of_norm] at h }
+
+ section dominated_convergence
+
+ variables {F : ℕ → α → β} {f : α → β} {bound : α → ℝ}
+
+ /-- `∥F n a∥ ≤ bound a` implies `ennreal.of_real ∥F n a∥ ≤ ennreal.of_real (bound a)` -/
+ private lemma of_real_F_le_bound (h : ∀ n, ∀ₘ a, ∥F n a∥ ≤ bound a) :
```

Comment:
In general, you should avoid marking lemmas as `private`, unless they are particular case of more general results that you prove below. The reason is that you never know what might be useful later on. If you want to hide them, you can put them in a namespace `dominated_convergence` for instance.